kube-runtime/src/watcher.rs

```diff
@@ -112,7 +112,10 @@ impl<K> Event<K> {
 /// The internal finite state machine driving the [`watcher`]
 enum State<K: Resource + Clone> {
     /// The Watcher is empty, and the next [`poll`](Stream::poll_next) will start the initial LIST to get all existing objects
-    Empty,
+    Empty {
+        continue_token: Option<String>,
+        objects: Vec<K>,
+    },
```

Comment:
Ah, you are storing the interim list state in memory in here. That's pretty clever.